src/sourceItemWrapper.js

```diff
@@ -663,32 +663,33 @@ class SourceItemWrapper extends ItemWrapper {
                 'loading',
                 Wikicite.getString('wikicite.source-item.add-identifier.progress.loading')
             );
-
             try {
                 if (identifiers.length > 0) {
-                    let citations = [];
                     await Zotero.Schema.schemaUpdatePromise;
-                    for (const identifier of identifiers) {
-                        let translation = new Zotero.Translate.Search();
+                    let citations = await Promise.all(identifiers.map(async (identifier) => {
+                        var translation = new Zotero.Translate.Search();
                         translation.setIdentifier(identifier);
+                        let translators = await translation.getTranslators();
+                        translation.setTranslator(translators);
```

Comment:
No, I couldn't find any information by following the git blame, but it was from 6 years ago so if it works without it, it should be fine?

Reply:
This block here seems to be checking, upon running `translate()`, whether translators have already been set. Otherwise, it seems to run `getTranslators()` and pass its result (via a `translators` handler) to `setTranslator()`. Therefore, it should be OK if we don't do it manually beforehand, I think.